beacon_node/store/src/lib.rs

```diff
@@ -185,7 +185,9 @@ pub trait Store<E: EthSpec>: Sync + Send + Sized + 'static {
 
 /// Reified key-value storage operation.  Helps in modifying the storage atomically.
 /// See also https://github.com/sigp/lighthouse/issues/692
-pub enum StoreOp {
+pub enum StoreOp<'a, E: EthSpec> {
+    PutBlock(SignedBeaconBlockHash, SignedBeaconBlock<E>),
+    PutState(BeaconStateHash, &'a BeaconState<E>),
```

Comment:
I'm on board with using a reference here, but with state summaries existing it might make sense to mix in the two variants from `StateBatch`:

```rust
// Allow owned states for the storage of intermediate states in block processing (uncommon)
PutState(BeaconStateHash, Cow<'a, BeaconState<E>>),
PutStateSummary(BeaconStateHash, HotStateSummary),
```

See https://github.com/sigp/lighthouse/blob/master/beacon_node/store/src/state_batch.rs

`StateBatch` could then be deleted of course.